kornia/augmentation/_2d/mix/cutmix.py

```diff
@@ -133,3 +133,124 @@ def apply_transform(  # type: ignore
+        >>> input = torch.rand(2, 1, 3, 3)
+        >>> input[0] = torch.ones((1, 3, 3))
+        >>> label = torch.tensor([0, 1])
+        >>> cutmix = RandomCutMix()
+        >>> cutmix(input, label)
+        (tensor([[[[0.8879, 0.4510, 1.0000],
+                  [0.1498, 0.4015, 1.0000],
+                  [1.0000, 1.0000, 1.0000]]],
+        <BLANKLINE>
+        <BLANKLINE>
+                [[[1.0000, 1.0000, 0.7995],
+                  [1.0000, 1.0000, 0.0542],
+                  [0.4594, 0.1756, 0.9492]]]]), tensor([[[0.0000, 1.0000, 0.4444],
+                 [1.0000, 0.0000, 0.4444]]]))
+    """
+
+    def __init__(
+        self,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        num_mix: int = 1,
+        cut_size: Optional[Union[torch.Tensor, Tuple[float, float]]] = None,
+        beta: Optional[Union[torch.Tensor, float]] = None,
+        same_on_batch: bool = False,
+        p: float = 1.0,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=1.0, p_batch=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        if height is not None or width is not None:
+            warnings.warn(
+                "height and width can be inferred automatically now. "
+                "The height and width arguments will be removed finally.",
+                category=DeprecationWarning,
+            )
+        self._param_generator = cast(rg.CutmixGenerator, rg.CutmixGenerator(cut_size, beta, num_mix, p=p))
+
+    def transform_labels(
+        self, label: torch.Tensor, params: Dict[str, torch.Tensor]
+    ) -> torch.Tensor:
+        height, width = params["image_shape"]
+        num_mixes = params["mix_pairs"].size(0)
+        batch_size = params["mix_pairs"].size(1)
+
+        _shape_validation(params["mix_pairs"], [num_mixes, batch_size], "mix_pairs")
+        _shape_validation(params["crop_src"], [num_mixes, batch_size, 4, 2], "crop_src")
+
+        out_labels = []
+        for pair, crop in zip(params["mix_pairs"], params["crop_src"]):
+            labels_permute = label.index_select(dim=0, index=pair.to(label.device))
+            w, h = infer_bbox_shape(crop)
+            lam = w.to(label.dtype) * h.to(label.dtype) / (width * height)  # width_beta * height_beta
+            out_labels.append(
+                torch.stack([label.to(label.dtype), labels_permute.to(label.dtype), lam.to(label.device)], dim=1)
+            )
+
+        return torch.stack(out_labels, dim=0)
+
+    def apply_transform(  # type: ignore
+        self, input: torch.Tensor, params: Dict[str, torch.Tensor]  # type: ignore
+    ) -> torch.Tensor:
+        height, width = input.size(2), input.size(3)
+        num_mixes = params["mix_pairs"].size(0)
+        batch_size = params["mix_pairs"].size(1)
+
+        _shape_validation(params["mix_pairs"], [num_mixes, batch_size], "mix_pairs")
+        _shape_validation(params["crop_src"], [num_mixes, batch_size, 4, 2], "crop_src")
+
+        out_inputs = input.clone()
```

Comment:
Won’t this be slow doing a copy ?

Reply:
Or we may offer an `inplace` argument to allow inplace permutation.

Reply:
not sure -- i'd avoid if possible out of place ops